File: core-examples/src/main/java/io/vertx/example/core/http/simpleform/SimpleFormServer.java
package io.vertx.example.core.http.simpleform;

import io.vertx.core.AbstractVerticle;
import io.vertx.launcher.application.VertxApplication;

/*
 * NOTE! It's recommended to use Vert.x-Web for examples like this
 *
 * @author <a href="http://tfox.org">Tim Fox</a>
 */
public class SimpleFormServer extends AbstractVerticle {

  public static void main(String[] args) {
    VertxApplication.main(new String[]{SimpleFormServer.class.getName()});
  }

  @Override
  public void start() throws Exception {
    vertx.createHttpServer().requestHandler(req -> {
      if (req.uri().equals("/")) {
        // Serve the index page
        req.response().sendFile("io/vertx/example/core/http/simpleform/index.html");
      } else if (req.uri().startsWith("/form")) {
        req.response().setChunked(true);
        req.setExpectMultipart(true);
        req.endHandler((v) -> {
          for (String attr : req.formAttributes().names()) {
            req.response().write("Got attr " + attr + " : " + req.formAttributes().get(attr) + "\n");
          }
          req.response().end();
        });
      } else {
        req.response().setStatusCode(404).end();
      }
    }).listen(8080);
  }
}

File: core-examples/src/main/kotlin/io/vertx/example/core/http/simpleform/SimpleFormServer.kt
import io.vertx.kotlin.common.json.*

class start : io.vertx.core.AbstractVerticle()  {
  override fun start() {
    vertx.createHttpServer().requestHandler({ req ->
      if (req.uri() == "/") {
        // Serve the index page
        req.response().sendFile("index.html")
      } else if (req.uri().startsWith("/form")) {
        req.response().setChunked(true)
        req.setExpectMultipart(true)
        req.endHandler({ v ->
          for (attr in req.formAttributes().names()) {
            req.response().write("Got attr ${attr} : ${req.formAttributes().get(attr)}\n")
          }
          req.response().end()
        })
      } else {
        req.response().setStatusCode(404).end()
      }
    }).listen(8080)
  }
}
